helix-rest/src/test/java/org/apache/helix/rest/metadatastore/accessor/TestZkRoutingDataReader.java

```diff
@@ -130,17 +116,11 @@ public void testGetRoutingDataMSRDChildEmptyValue() {
     }
   }
 
-  private void deleteRoutingDataPath() throws Exception {
-    Assert.assertTrue(TestHelper.verify(() -> {
-      ZK_SERVER_MAP.get(ZK_ADDR).getZkClient()
-          .deleteRecursively(MetadataStoreRoutingConstants.ROUTING_DATA_PATH);
-
-      if (ZK_SERVER_MAP.get(ZK_ADDR).getZkClient()
-          .exists(MetadataStoreRoutingConstants.ROUTING_DATA_PATH)) {
-        return false;
-      }
-
-      return true;
-    }, TestHelper.WAIT_DURATION), "Routing data path should be deleted after the tests.");
+  private void clearRoutingDataPath() {
+    for (String zkRealm : _baseAccessor
+        .getChildNames(MetadataStoreRoutingConstants.ROUTING_DATA_PATH, AccessOption.PERSISTENT)) {
+      _baseAccessor.remove(MetadataStoreRoutingConstants.ROUTING_DATA_PATH + "/" + zkRealm,
+          AccessOption.PERSISTENT);
```

Comment:
Nit: Outside of helix-core module, we should try to use ZkClient instead of BaseDataAccessor.

Reply:
I believe we talked about this before implementing and decided to use `BaseDataAccessor`. This is good to know. 
Regardless, how do you feel about using `HelixZkClient`? It's used in other tests as well (like the one for leader election), but it's a deprecated class. Is it really okay to use it among the tests? I'd expect the general consensus to be avoid using it.

Reply:
Resolved offline. Using `HelixZkClient` and will migrate later.